### PR TITLE
Fixing incorrect bulk request took time

### LINKED ISSUE #111854: Bulk API returns the very big `took` value instead milliseconds
### Elasticsearch Version

8.15

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

GNU/Linux

### Problem Description

The Bulk API returns a huge took value in 8.15, example:
```json
{
    "errors": false,
    "took": 6641016797,
    "items": [
        {
            "index": {
                "_index": "index-abc17",
                "_id": "2323578",
                "_version": 5,
                "result": "updated",
                "_shards": {
                    "total": 3,
                    "successful": 3,
                    "failed": 0
                },
                "_seq_no": 79341,
                "_primary_term": 11,
                "status": 200
            }
        }
    ]
}
```
although according to the documentation it is integer (32 bit).  
> took
(integer) How long, in milliseconds, it took to process the bulk request.

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html

This causes the Jackson parse exception in our internal library (the official Elastic `ElasticsearchClient` uses `long` internally).
Even if I fix it to be `long` in our library the value seems to be odd: something completed within `6,641,016,797` ms? 

### Steps to Reproduce

In the DevConsole:

1.  
  ```
  PUT magic-index
  {
    "mappings": {
      "properties": {
        "user_number": {
          "type": "keyword"
        }, 
        "active_time": {
          "type": "date"
        }, 
        "keywords": {
          "type": "keyword"
        }
      }
    }
  }
  ```

2.
  ```
  POST _bulk
  {"index":{"_id":"32","_index":"magic-index"} }
  {"user_number": "213343334", "active_time": "2024-11-15T16:50:14.167-04:00", "keywords": "alpha"}
  ```

3. 
the result is 

  ```json
  {
    "errors": false,
    "took": 9430291,
    "items": [
      {
        "index": {
          "_index": "magic-index",
          "_id": "32",
          "_version": 1,
          "result": "created",
          "_shards": {
            "total": 2,
            "successful": 2,
            "failed": 0
          },
          "_seq_no": 0,
          "_primary_term": 1,
          "status": 201
        }
      }
    ]
  }
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by masseyke**
In #109889 TransportAbstractBulkAction was using `ThreadPool::relativeTimeInMillis` to get the start time, and then subtracting that from the nano representation of the finish time to compute the took time. So the took time was orders of magnitude larger than it should have been.
Closes #111854

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/bulk/TransportAbstractBulkAction.java [hunk 1]
```diff
@@ -321,9 +321,9 @@ private void applyPipelinesAndDoInternalExecute(
         Executor executor,
         ActionListener<BulkResponse> listener
     ) {
-        final long relativeStartTime = threadPool.relativeTimeInMillis();
+        final long relativeStartTimeNanos = threadPool.relativeTimeInNanos();
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/bulk/TransportAbstractBulkAction.java:324]: Is there a good reason not to use `relativeTime()` here?

**Review by DaveCTurner — COMMENTED** [review]
I thoroughly approve of adding a `millis` or `nanos` suffix to symbol names that return a time as a bare `long`. Could we propagate that change further to cover all applicable symbols in this area? E.g. `relativeTimeProvider -> relativeTimeNanosProvider` etc.

**Comment by masseyke** [comment]
> I thoroughly approve of adding a `millis` or `nanos` suffix to symbol names that return a time as a bare `long`. Could we propagate that change further to cover all applicable symbols in this area? E.g. `relativeTimeProvider -> relativeTimeNanosProvider` etc.

Yeah I'll do this. I'm currently reworking it to use millis instead of nanos because `org.elasticsearch.common.util.concurrent.DeterministicTaskQueue`'s [ThreadPool](https://github.com/elastic/elasticsearch/blob/main/test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java#L342) does not implement `relativeTimeInNanos`. I don't think there was any benefit to switching to nano resolution here.

**Comment by DaveCTurner** [comment]
You could reasonably implement `relativeTimeInNanos` on the DTQ threadpool as just `currentTimeMillis * 1_000_000`. It's only not implemented because nobody needed it yet.

**Comment by masseyke** [comment]
That would definitely be easier.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**